storagetransfer/api/StorageTransfer.Samples.Tests/StorageFixture.cs

```diff
@@ -97,16 +168,47 @@ public void Dispose()
             }
             catch (Exception)
             {
-                // Do nothing, we delete on a best effort basis.
+                // If bucket is not empty, we delete on a best effort basis.
+                foreach (var storageObject in Storage.ListObjects(BucketNameSink, ""))
+                {
+                    Storage.DeleteObject(BucketNameSink, storageObject.Name);
+                }
+                Storage.DeleteBucket(BucketNameSink);
             }
             try
             {
                 Storage.DeleteBucket(BucketNameSource);
             }
             catch (Exception)
             {
-                // Do nothing, we delete on a best effort basis.
+                // If bucket is not empty, we delete on a best effort basis.
+                foreach (var storageObject in Storage.ListObjects(BucketNameSource, ""))
+                {
+                    Storage.DeleteObject(BucketNameSource, storageObject.Name);
+                }
+                Storage.DeleteBucket(BucketNameSource);
+            }
+
+            try
+            {
+                TopicName topicName = TopicName.FromProjectTopic(ProjectId, TopicId);
+                PublisherClient.DeleteTopic(topicName);
             }
+            catch (RpcException ex)
+            {
+                throw new Exception($"Exception occur while deleting Topic {TopicId} Exception: {ex}");
```

Comment:
We don't rethrow the exception, that will make dispose fail and we don't want that. We delete resources on a best effort basis.